runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -739,13 +739,14 @@ function M.get_virtual_text_chunks_for_line(bufnr, line, line_diags, opts)
   end
   local last = line_diags[#line_diags]
 
-  -- TODO(tjdevries): Allow different servers to be shown first somehow?
-  -- TODO(tjdevries): Display server name associated with these?
   if last.message then
+    -- get server name
+    local server_name = vim.lsp.get_active_clients()[1].config.cmd[1]:match('[^\\/]+$')
```

Comment:
`vim.lsp.get_active_clients()[1]` this just assumes the messages comes from the first server.
What if the messages is from a different server?

Reply:
@lukas-reineke yep. you are right.

Reply:
Also, don't they have a name param? I don't remember off hand.

Reply:
some server  command name will contain the path like `sumneko/lua-language-server`. As config value. if user turn off virtual text. this will don't show right? so need set this as config value too?